src/extensionVariables.ts

```diff
@@ -4,13 +4,21 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AzExtTreeDataProvider, AzExtTreeItem, IAzExtOutputChannel } from "@microsoft/vscode-azext-utils";
-import { DiagnosticCollection, Disposable, ExtensionContext, TreeView } from "vscode";
+import { DiagnosticCollection, Disposable, Event, EventEmitter, ExtensionContext, TreeView } from "vscode";
 import { ActivityLogTreeItem } from "./activityLog/ActivityLogsTreeItem";
 import { AppResourceProvider } from "./api";
 import { TagFileSystem } from "./commands/tags/TagFileSystem";
 import { AzureAccountTreeItem } from "./tree/AzureAccountTreeItem";
 import { ExtensionActivationManager } from "./utils/ExtensionActivationManager";
 
+namespace extEmitters {
```

Comment:
I like this idea!

Reply:
One thing to note here is that the order of the declaring the namespaces matters. I had them declared below `ext`, and TS didn't yell at me but they were always undefined at runtime.